.ci/test-install-docs.sh

```diff
@@ -6,18 +6,239 @@
 
 set -e
 
+# The go binary isn't installed, but we checkout the repos to the standard
+# golang locations.
+export GOPATH=${GOPATH:-${HOME}/go}
+
+typeset -r script_name="${0##*/}"
+typeset -r script_dir="$(cd "$(dirname "${0}")" && pwd)"
+
+typeset -r docker_image="busybox"
+typeset -r kata_project_url="github.com/kata-containers"
+typeset -r test_repo="${kata_project_url}/tests"
+typeset -r test_repo_url="https://${test_repo}"
+typeset -r test_repo_dir="${GOPATH}/src/${test_repo}"
+typeset -r kata_project_dir="${GOPATH}/src/${kata_project_url}"
+
+typeset -r mgr="${test_repo_dir}/cmd/kata-manager/kata-manager.sh"
+typeset -r doc_to_script="${test_repo_dir}/.ci/kata-doc-to-script.sh"
+
+die()
+{
+	local msg="$*"
+	echo >&2 "ERROR: $msg"
+	exit 1
+}
+
 info()
 {
-	msg="$*"
+	local msg="$*"
 	echo "INFO: $msg"
 }
 
-# Detect if any installation documents changed. If so, run the
-# kata manager to "execute" the install guide to ensure the
-# commands it specified result in a working system.
-check_install_guides()
+usage()
+{
+	cat <<EOT
+Description: Run Kata documentation CI tests.
+
+Usage: $script_name [options]
+
+Options:
+
+ -h        : Show this help.
+ -t <dir>  : Run all scripts ("\*.sh" files) in the specified
+             directory.
+
+Notes:
+  - The '-t' option is not generally useful - it is used by this
+    script which re-exec's itself with this option.
+
+EOT
+}
+
+# Re-execute the running script from a temporary directory to allow the
+# script to continue executing even if the original source file is deleted.
+reexec_in_tmpdir()
+{
+	local -r test_dir="$1"
+
+	[ -d "${test_dir}" ] || die "invalid test dir: ${test_dir}"
+
+	if [ "${script_dir}" = "${test_dir}" ]
+	then
+		# Already running from temp directory so nothing else to do
+		return
+	fi
+
+	local new
+	new="${test_dir}/${script_name}"
+
+	install --mode 750 "${0}" "${new}"
+
+	info "Re-execing ${0} as ${new}"
+
+	cd "${test_dir}"
+
+	exec "${new}" -t "${test_dir}/tests"
+}
+
+# Grab a copy of the tests repository
+get_tests_repo()
+{
+	[ -d "${test_repo_dir}" ] && return
+
+	mkdir -p "${kata_project_dir}"
+
+	git clone "${test_repo_url}" "${test_repo_dir}"
+}
+
+# Delete all local github repo clones.
+#
+# This is required to ensure that the tests themselves (re-)create these
+# clones.
+delete_kata_repos()
+{
+	[ -n "${KATA_DEV_MODE}" ] && die "Not continuing as this is a dev system"
+	[ -z "${CI}" ] && die "Not continuing as this is a non-CI environment"
+
+	local cwd="$PWD"
+
+	info "Deleting all local kata repositories below ${kata_project_dir}"
```

Comment:
This should be a warning with a big red button :D

Reply:
As shown in the code, we only do this if running under a CI and developer mode is not enabled.